### PR TITLE
Fix integer/float errors in survey

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
Fix console errors for integer & float default values when setting up a template survey

addresses #8331 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/util/validators.jsx [hunk 1]
```diff
@@ -76,6 +77,16 @@ export function integer(i18n) {
   };
 }
 
+export function number(i18n) {
+  return value => {
+    const str = String(value);
+    if (/^[0-9]*(\.[0-9]*)?$/.test(str)) {
```
Comment (jakemcdermott) [inline awx/ui_next/src/util/validators.jsx]: Probably should also handle:
  - negative numbers `/^[0-9]*(\.[0-9]*)?$/.test(String(-2.2))`
  - big numbers `/^[0-9]*(\.[0-9]*)?$/.test(String(999999999999999999999.9))`

File: awx/ui_next/src/util/validators.jsx [hunk 2]
```diff
@@ -68,6 +68,7 @@ export function noWhiteSpace(i18n) {
 
 export function integer(i18n) {
   return value => {
+    console.log(value);
```
Comment (jakemcdermott) [inline awx/ui_next/src/util/validators.jsx]:
```suggestion
```

**Review by jakemcdermott — COMMENTED** [review]
lgtm, just a few nits & I think we should merge this.

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
lgtm, just a few nits & I think we should test/merge this.

**Review by jakemcdermott — APPROVED** [review]

**Comment by tiagodread** [comment]
@keithjgrant I tried to create a float survey and got this result:

survey

Also, try add one each survey type into JT, when you add Multiple choice option and try launch the JT the page crashes totally:

survey1

**Comment by keithjgrant** [comment]
> @keithjgrant I tried to create a float survey and got this result:

I get this error in the old UI as well... I think this should be filed as an API bug. It's not accepting `2` as a value

> Also, try add one each survey type into JT, when you add Multiple choice option and try launch the JT the page crashes totally:

Should be fixed now

**Comment by unlikelyzero** [comment]
please check this issue at the same time https://github.com/ansible/awx/issues/8222

**Comment by unlikelyzero** [comment]
@keithjgrant @tiagodread please check #8222 and also file the float bug against the API

**Comment by tiagodread** [comment]
@keithjgrant I saw that the ui_next and old UI is passing the `default` as integer:

```
{
    "description": "",
    "name": "",
    "spec": [
        {
            "choices": "",
            "default": 2,
            "max": 1024,
            "min": 0,
            "new_question": true,
            "question_description": "",
            "question_name": "1.0+1.0",
            "required": true,
            "type": "float",
            "variable": "float"
        }
    ]
}
```

changing 2 to 2.0 the API can save successfully

**Comment by keithjgrant** [comment]
JavaScript doesn't distinguish between 2 and 2.0. I might be able to send the string "2.0", but a string is definitely not a float

**Comment by keithjgrant** [comment]
@tiagodread JavaScript doesn't distinguish between 2 and 2.0. The only other option I have is to send the string "2.0", but a string is definitely not a float, and the API returns the same error.

**Comment by tiagodread** [comment]
Open a new issue to fix only Float default value here https://github.com/ansible/awx/issues/8460

**Review by tiagodread — APPROVED** [review]
Good Job!
Tests covered by 5635

**Pull request merged**